docs/docs/apis/core/address-api.md

```diff
@@ -0,0 +1,338 @@
+---
+title: "Address API"
+description: "Address generation, validation, and conversion utilities"
+sidebar_position: 3
+---
+
+# Address API
+
+The Address API provides comprehensive functionality for working with Cardano addresses. It supports address creation, validation, conversion, and extraction of address components. The API handles both Shelley addresses (current) and Byron addresses (legacy) with full support for different address types and networks.
+
+## Key Features
+
+- **Address Creation**: Create addresses from various inputs (strings, bytes, credentials)
+- **Address Validation**: Validate address format and network compatibility
+- **Address Conversion**: Convert between different address formats and representations
+- **Credential Extraction**: Extract payment and delegation credentials from addresses
+- **Network Detection**: Determine network (mainnet/testnet) from address
+- **Type Support**: Support for all Cardano address types (base, enterprise, pointer, reward)
+
+## Core Classes
+
+### Address Class
+The main class for Shelley address operations, supporting Bech32 encoded addresses and credential extraction.
+
+### AddressProvider Class
+Utility class for generating addresses from accounts, keys, and scripts.
+
+### AddressUtil Class
+Utility class for address validation, conversion, and manipulation operations.
+
+**Key Methods:**
```

Comment:
This section is little confusing as it doesn't clearly mention the class for a method. For example: toBech32(), getPaymentCredential() etc.